### ISSUE TITLE
Modelica.Math.Vectors.normalize contains smooth(0,noEvent(...))

### ISSUE THREAD

**Issue by modelica-trac-importer** [modelica/Modelica#1759]
**Reported by stefanv on 11 Aug 2015 19:10 UTC**
To the best of my knowledge, `smooth` and `noEvent` have no meaning inside functions.

----

Migrated-From: https://trac.modelica.org/Modelica/ticket/1759

**Comment by modelica-trac-importer** [comment]
**Modified by beutlich on 18 Aug 2015 12:25 UTC**

**Comment by modelica-trac-importer** [comment]
**Comment by otter on 28 Sep 2015 07:16 UTC**
Replying to [ticket:1759 stefanv]:
> To the best of my knowledge, `smooth` and `noEvent` have no meaning inside functions.

Fixed in cf313732d73f9f8f30e4b404885e76cbc0441c92:
Added the following comment in the code (trunk):
```
 This function has the inline annotation. If the function is inlined:
     - "smooth(..)" defines how often the expression can be differentiated
       (if symbolic processing is performed). 
     - "noEvent(..)" prevents event handling of the if-clause,
       in order to guard against division by zero.
```

**Comment by modelica-trac-importer** [comment]
**Modified by otter on 28 Sep 2015 07:17 UTC**

**Comment by modelica-trac-importer** [comment]
**Comment by stefanv on 28 Sep 2015 13:48 UTC**
But this means that the semantics of a function are different depending on whether or not it is successfully inlined. That should not be the case, and a tool that inlines a function should automatically add smooth and noEvent where needed.

**Comment by modelica-trac-importer** [comment]
**Comment by sjoelund&#46;se on 28 Sep 2015 20:46 UTC**
Replying to [comment:4 stefanv]:
> But this means that the semantics of a function are different depending on whether or not it is successfully inlined. That should not be the case, and a tool that inlines a function should automatically add smooth and noEvent where needed.

Functions generate events when GenerateEvents=true. But it is not set here.

**Comment by modelica-trac-importer** [comment]
**Comment by otter on 13 Dec 2015 06:54 UTC**
Fixed in 207cb79b1b21a36cb72fdd194776fe372fda2925 by
removed noEvent (had no effect since inlining a function adds automatically noEvent).

Replying to [comment:4 stefanv]:
> But this means that the semantics of a function are different depending on whether or not it is successfully inlined. That should not be the case, and a tool that inlines a function should automatically add smooth and noEvent where needed.

A tool cannot automatically add "smooth" because in general it cannot figure out how "smooth" an if-expression is.

If the function is not inlined, "smooth(..)" has no effect. If it is inlined, it gives additional information to the Modelica tool. The tool may or may not use this information (smooth(..) is defined in this way).

**Comment by modelica-trac-importer** [comment]
**Changelog modified by otter on 13 Dec 2015 06:54 UTC**
Math.Vectors.normalize: Removed unnecessary noEvent(..)
